src/Symfony/Component/Security/Csrf/TokenStorage/SessionTokenStorage.php

```diff
@@ -45,62 +63,78 @@ public function __construct(SessionInterface $session, string $namespace = self:
      */
     public function getToken(string $tokenId)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        $session = $this->getSession();
+        if (!$session->isStarted()) {
+            $session->start();
         }
 
-        if (!$this->session->has($this->namespace.'/'.$tokenId)) {
+        if (!$session->has($this->namespace.'/'.$tokenId)) {
             throw new TokenNotFoundException('The CSRF token with ID '.$tokenId.' does not exist.');
         }
 
-        return (string) $this->session->get($this->namespace.'/'.$tokenId);
+        return (string) $session->get($this->namespace.'/'.$tokenId);
     }
 
     /**
      * {@inheritdoc}
      */
     public function setToken(string $tokenId, string $token)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        $session = $this->getSession();
+        if (!$session->isStarted()) {
+            $session->start();
         }
 
-        $this->session->set($this->namespace.'/'.$tokenId, $token);
+        $session->set($this->namespace.'/'.$tokenId, $token);
     }
 
     /**
      * {@inheritdoc}
      */
     public function hasToken(string $tokenId)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        $session = $this->getSession();
+        if (!$session->isStarted()) {
+            $session->start();
         }
 
-        return $this->session->has($this->namespace.'/'.$tokenId);
+        return $session->has($this->namespace.'/'.$tokenId);
     }
 
     /**
      * {@inheritdoc}
      */
     public function removeToken(string $tokenId)
     {
-        if (!$this->session->isStarted()) {
-            $this->session->start();
+        $session = $this->getSession();
+        if (!$session->isStarted()) {
+            $session->start();
         }
 
-        return $this->session->remove($this->namespace.'/'.$tokenId);
+        return $session->remove($this->namespace.'/'.$tokenId);
     }
 
     /**
      * {@inheritdoc}
      */
     public function clear()
     {
-        foreach (array_keys($this->session->all()) as $key) {
+        $session = $this->getSession();
+        foreach (array_keys($session->all()) as $key) {
             if (0 === strpos($key, $this->namespace.'/')) {
-                $this->session->remove($key);
+                $session->remove($key);
             }
         }
     }
+
+    private function getSession(): SessionInterface
+    {
+        try {
+            return $this->requestStack->getSession();
+        } catch (SessionNotFoundException $e) {
+            trigger_deprecation('symfony/security-csrf', '5.3', 'Using the "%s" without a session has no effect and is deprecated. It will throw an exception in Symfony 6.0', __CLASS__);
```

Comment:
missing final dot

Reply:
we should tell which exception will be thrown, to that ppl can prepare their code
also, we should add the throwing line as a comment, to ease cleaning this deprecation for v6
last: this misses UPGRADE+CHANGELOG entries

Reply:
done